### ISSUE TITLE
Iterative test case with multiple Snapshot.Match fails in 0.5.4 but worked in 0.5.2

### ISSUE THREAD

**Issue by moredatapls** [SwissLife-OSS/snapshooter#98]
**Describe the bug**

We have a test case that consists of a for loop. In each iteration, an operation is performed and its result is compared to a snapshot. The name of the snapshot is dynamically created in each loop.

```cs
[Fact]
public void IterativeTest()
{
    const string nameBase = "TestCase";

    for (var i = 0; i < 2; i++)
    {
        var contents = $"this is iteration = {i}";

        var name = $"{nameBase}_{i}";

        Snapshot.Match(contents, Snapshot.FullName(SnapshotNameExtension.Create(name)));
    }
}
```

When I run this test with version 0.5.2, the files are created and compared as expected. When I upgrade to 0.5.4, the test case fails because in the second iteration, the contents are compared to the file of the first iteration.

In 0.5.2

* `i = 0`: `name = "TestCase_0"`, contents are compared to `TestCase_0.snap` -> successful
* `i = 1`: `name = "TestCase_1"`, contents are compared to `TestCase_1.snap` -> successful
* -> test succeeds

In 0.5.4

* `i = 0`: `name = "TestCase_0"`, contents are compared to `TestCase_0.snap` -> successful
* `i = 1`: `name = "TestCase_1"`, contents are compared to `TestCase_0.snap` -> fails
* -> test fails

**To Reproduce**

1. Add the test case I posted above
2. Run with 0.5.2, will work
3. Run with 0.5.4, will fail

**Expected behavior**

The test case should not fail.

**Desktop (please complete the following information):**

 - OS: macOS Catalina
 - Version: 10.15.5

**Comment by michaelstaib** [comment]
Hi,

I have introduced this little bug with a feature that I have added to support some cases with test helpers.

We will introduce a new test context and with that fix this issue.

Thanks for reporting this.

**Comment by jannickj** [comment]
maybe remove the bug until feature is ready?, I just spent 20 min debugging my code :(

**Comment by RohrerF** [comment]
this issue should be fixed in version 0.5.5
